### ISSUE TITLE
Announce Spring Boot upgrade to version 3.0 and related changes to consider

### ISSUE THREAD

**Issue by romansmirnov** [camunda/zeebe#12044]
**Description**

With 8.2, Zeebe comes with Spring Boot 3. The most noteworthy change in Spring is the URL matching handling: https://github.com/spring-projects/spring-boot/wiki/Spring-Boot-3.0-Migration-Guide#spring-mvc-and-webflux-url-matching-changes For example, `http://localhost:9600/actuator/rebalance/ will fail with a `404` while `http://localhost:9600/actuator/rebalance` will succeed. Please note, this only applies to the actuator and Zeebe's management endpoints (gRPC is not affected by that change).

* Announce Dependency Spring Boot 3 upgrade: https://docs.camunda.io/docs/reference/announcements/
* Update Guide 8.1 to 8.2: Describe URL matching handling change, i.e., users must remove any trailing slashes when using the actuator or management endpoints.

---

related to https://github.com/camunda/zeebe/pull/11103
related to https://github.com/camunda/camunda-platform-docs/pull/1803

**Comment by korthout** [comment]
ZPA Triage: We think that these endpoints are a ZDP responsibility. Removing this issue from our board

**Comment by Sijoma** [comment]
Interesting one. This seems to break OpenAPI compatibility on the first glance. Although the spec defines the 
url as `{schema}://{host}:{port}/actuator/backups` for backups. 

The generated client with [openapi-generator ](https://openapi-generator.tech/docs/configuration/) seems to include a trailing slash as I'm not able to create backups anymore with a test build (for GCS) that @oleschoenburg  supplied to me.

**Comment by romansmirnov** [comment]
@Sijoma, did you manage to get the backups working?

**Comment by Sijoma** [comment]
I've scrapped the idea to generate the clients directly with the OpenAPI specification. 

The openapi-generator somehow was not flexible with the adjustment on the trailing slash. I only managed to get the client working after editing the generated client -> https://github.com/Sijoma/camunda-backup-clients/commit/64737a232b9c2e9943f53ecd981f0d77d7f17dae .

Interesting that Spring Boot 3 introduces this change.
